komodo/yaml_file_types.py

```diff
@@ -38,16 +133,544 @@ class ManifestFile(YamlFile):
+
+                importance_errors = Package.validate_package_importance_with_errors(
+                    package_name, status.get("importance")
+                )
+                errors.extend(importance_errors)
+
+        handle_validation_errors(errors, message)
+
+
+class Package:
+    VALID_VISIBILITIES = ["public", "private"]
+    VALID_IMPORTANCES = ["low", "medium", "high"]
+    VALID_MATURITIES = ["experimental", "stable", "deprecated"]
+    VALID_MAKES = ["rpm", "cmake", "sh", "pip", "rsync", "noop", "download"]
+
+    @staticmethod
+    def validate_package_name(package_name: str) -> bool:
+        if isinstance(package_name, str):
+            if str.islower(package_name):
+                return True
+            raise ValueError(f"Package name '{package_name}' should be lowercase.")
+        raise TypeError(f"Package name ({package_name}) should be of type string")
+
+    @staticmethod
+    def validate_package_version(package_name: str, package_version: str) -> bool:
+        if isinstance(package_version, str):
+            return True
+        raise TypeError(
+            f"Package '{package_name}' has invalid version type ({package_version})"
+        )
+
+    @staticmethod
+    def validate_package_entry(package_name: str, package_version) -> bool:
+        Package.validate_package_name(package_name)
+        Package.validate_package_version(package_name, package_version)
+
+    @staticmethod
+    def validate_package_entry_with_errors(
+        package_name: str, package_version: str
+    ) -> List[str]:
+        """
+        Validates package name and version, and returns a list of error messages
+        """
+        errors = []
+        try:
+            Package.validate_package_entry(package_name, package_version)
+        except (ValueError, TypeError) as e:
+            errors.append(str(e))
+        return errors
+
+    @staticmethod
+    def validate_package_importance(package_name: str, package_importance: str) -> bool:
+        if isinstance(package_importance, str):
+            if package_importance in Package.VALID_IMPORTANCES:
+                return True
+            raise ValueError(
+                f"{package_name} has invalid importance value ({package_importance})"
+            )
+        raise TypeError(
+            f"{package_name} has invalid importance type ({package_importance})"
+        )
+
+    @staticmethod
+    def validate_package_importance_with_errors(
+        package_name, package_importance: str
+    ) -> List[str]:
+        """
+        Validates package importance of a package and returns a list of error messages
+        """
+        errors = []
+        try:
+            Package.validate_package_importance(package_name, package_importance)
+        except (ValueError, TypeError) as e:
+            errors.append(str(e))
+        return errors
+
+    @staticmethod
+    def validate_package_visibility(package_name: str, package_visibility: str) -> None:
```

Comment:
(here the typing is wrong, it says None, but there is a `return True` statement)